.github/workflows/kong3.0_compatibility.yaml

```diff
@@ -86,8 +88,103 @@ jobs:
+  build-push-e2e-images:
+    environment: "Configure ci"
+    runs-on: ubuntu-latest 
+    steps:
+    - name: setup golang
+      uses: actions/setup-go@v3
+      with:
+        go-version: '^1.19'
+
+    - name: cache go modules
+      uses: actions/cache@v3
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-build-codegen-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-build-codegen-
+
+    - name: checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+      with:
+        driver-opts: network=host
+
+    - name: Build and push to local registry
+      id: docker_build
+      uses: docker/build-push-action@v3
+      with:
+        context: .
+        load: true
+        file: Dockerfile
+        tags: kong/kubernetes-ingress-controller:ci
+        target: distroless
+    
+  e2e-tests-oss:
+    environment: "Configure ci"
+    runs-on: ubuntu-latest
+    needs: build-push-e2e-images
+    steps:
+    - name: run non-enterprise cases of e2e test
+      run: make test.e2e
+      ## run tests without KONG_LICENSE_DATA env to skip enterprise tests.
+      env:
+        TEST_KONG_CONTROLLER_IMAGE_LOAD: kong/kubernetes-ingress-controller:ci
+        TEST_KONG_CONTROLLER_IMAGE_OVERRIDE: kong/kubernetes-ingress-controller:ci
+        TEST_KONG_IMAGE_LOAD: kong/kong:3.0.0-alpha.14-alpine
+        KONG_CLUSTER_VERSION: "v1.24.2"
+        ISTIO_VERSION: "v1.14.1"
+        NCPU: 1
+
+    - name: upload diagnostics for e2e OSS 
+      if: ${{ always() }}
+      uses: actions/upload-artifact@v3
+      with:
+        name: diagnostics-e2e-tests-oss
+        path: /tmp/ktf-diag*
+        if-no-files-found: ignore
+
+  e2e-tests-enterprise:
+    environment: "Configure ci"
+    runs-on: ubuntu-latest
+    needs: build-push-e2e-image
+    steps:
+    - name: setup kong enterprise license
+      uses: Kong/kong-license@master
+      id: license
+      with:
+        # PULP_PASSWORD secret is set in "Configure ci" environment
+        password: ${{ secrets.PULP_PASSWORD }}
+
+    - name: run enterprise cases of e2e test
+      run: make test.e2e
+      env:
+        TEST_KONG_CONTROLLER_IMAGE_LOAD: kong/kubernetes-ingress-controller:ci
+        TEST_KONG_CONTROLLER_IMAGE_OVERRIDE: kong/kubernetes-ingress-controller:ci
+        TEST_KONG_IMAGE_LOAD: kong/kong-gateway-internal:3.0.x.x-alpine
+        KONG_LICENSE_DATA: ${{ steps.license.outputs.license }}
+        TEST_KONG_PULL_USERNAME: ${{ secrets.GHA_DOCKERHUB_PULL_USER }}
+        TEST_KONG_PULL_PASSWORD: ${{ secrets.GHA_KONG_ORG_DOCKERHUB_PUBLIC_TOKEN }}
+        KONG_CLUSTER_VERSION: "v1.24.2"
+        ISTIO_VERSION: "v1.14.1"
+        NCPU: 1
+
+    - name: upload diagnostics for e2e enterprise 
+      if: ${{ always() }}
```

Comment:
Why not when the test fail?

Reply:
when test passed, their would be no such files and actually no files will be updated.